webapp/src/components/systemConsole/surveyResults/surveyResults.tsx

```diff
@@ -12,74 +12,45 @@ import type {SurveyResult} from 'types/plugin';
-                surveyId: utils.uuid(),
-                startDate: '01/01/2024',
-                endDate: '31/01/2024',
-                npsScore: 0,
-                receiptsCount: 950,
-                responseCount: 24,
-                status: 'in_progress',
-            },
-            {
-                surveyId: utils.uuid(),
-                startDate: '01/10/2023',
-                endDate: '30/10/2023',
-                npsScore: 9.1,
-                receiptsCount: 539,
-                responseCount: 328,
-                status: 'ended',
-            },
-            {
-                surveyId: utils.uuid(),
-                startDate: '01/07/2023',
-                endDate: '5/07/2023',
-                npsScore: 8.8,
-                receiptsCount: 496,
-                responseCount: 196,
-                status: 'ended',
-            },
-            {
-                surveyId: utils.uuid(),
-                startDate: '01/01/2023',
-                endDate: '31/01/2023',
-                npsScore: 7.5,
-                receiptsCount: 950,
-                responseCount: 24,
-                status: 'ended',
-            },
-            {
-                surveyId: utils.uuid(),
-                startDate: '31/01/2023',
-                endDate: '20/02/2023',
-                npsScore: 8.9,
-                receiptsCount: 1542,
-                responseCount: 1104,
-                status: 'ended',
-            },
-        ];
+    const [surveyResults, setSurveyResults] = useState<SurveyResult[]>([]);
+
+    const calculateNPS = (promoters: number, passives: number, detractors: number): number => {
+        const totalResponses = promoters + passives + detractors;
+
+        const promoterPercentage = (promoters / totalResponses) * 100;
+        const detractorPercentage = (detractors / totalResponses) * 100;
+
+        return Math.round(promoterPercentage - detractorPercentage);
     };
 
-    const [surveyResults, setSurveyResults] = useState<SurveyResult[]>([]);
+    const hydrateSurveyResults = useCallback((surveyResults: SurveyResult[]) => {
+        surveyResults.forEach((surveyResult) => {
+            surveyResult.startDate = new Date(surveyResult.startTime);
 
-    useEffect(() => {
-        const results = dummySurveyResults();
-        results.sort((a, b) => {
-            const aStartDate = parse(a.startDate, 'dd/MM/yyyy', new Date()).getSeconds();
-            const bStartDate = parse(b.startDate, 'dd/MM/yyyy', new Date()).getSeconds();
-            return aStartDate - bStartDate;
+            surveyResult.endDate = new Date(surveyResult.startTime);
+            surveyResult.endDate.setDate(surveyResult.endDate.getDate() + surveyResult.duration);
+
+            surveyResult.npsScore = calculateNPS(surveyResult.promoterCount, surveyResult.passiveCount, surveyResult.detractorCount);
         });
-        setSurveyResults(results);
     }, []);
 
+    useEffect(() => {
+        const work = async () => {
```

Comment:
can we be more descriptive on what `work` means?

Reply:
Done. Renamed.